src/hooks/useGridDimensions.ts

```diff
@@ -0,0 +1,38 @@
+import { useRef, useState, useLayoutEffect } from 'react';
+
+// https://github.com/microsoft/TypeScript/issues/37861
+interface ResizeObserverEntry {
+  contentRect: {
+    width: number;
+    height: number;
+  };
+}
+
+type ResizeObserver = new (callback: (entries: readonly ResizeObserverEntry[]) => void) => {
+  observe: (target: Element) => void;
+  disconnect: () => void;
+};
+
+export function useGridDimensions(): [React.RefObject<HTMLDivElement>, number, number] {
+  const gridRef = useRef<HTMLDivElement>(null);
+  const [gridWidth, setGridWidth] = useState(1);
+  const [gridHeight, setGridHeight] = useState(1);
+
+  useLayoutEffect(() => {
+    const { ResizeObserver } = window as typeof window & { ResizeObserver: ResizeObserver };
```

Comment:
I think it's fair to ask devs to use a polyfill to get the grid working, though that'd only be on IE11/old Edge/Safari 13.
Chrome and Firefox have had it for a long time now.
I can add a `if (!ResizeObserver) return;` if you want, but the grid will look quite empty.

Reply:
May be we can add a note on the readme file that a polyfill is required for old browsers. Something like https://reactjs.org/docs/javascript-environment-requirements.html

Reply:
I've added a line in the readme.